src/components/partials/UserContext.tsx

```diff
@@ -0,0 +1,39 @@
+import { createContext, useState } from 'react'
+
+type userType = {
+  username: string
+  auth: boolean
+}
+
+export type UserContextType = {
+  user: userType
+  login: (username: string) => void
+  logout: () => void
+}
+
+const UserContext = createContext<UserContextType | null>(null)
+
+const UserProvider = ({ children }: { children: React.ReactElement }) => {
+  // User is the name of the "data" that gets stored in context
+  const [user, setUser] = useState<userType>({ username: 'kitty', auth: false })
+
+  // Login updates the user data with a name parameter
+  const login = (username: string) => {
```

Comment:
the term login and logout seems to mislead. I'd suggest just leave as setUser(user: userModel) and return setUser and probably clearUser instead  of logout()

Reply:
So the User model includes alot of fields like (first name, last name, password, email, username) , so Im thinking of just using the LoginPayload model that just has (username, token) which i think might change based on Cognito later but since mainly we just need username, I'm thinking it's better? Let me know if otherwise

Reply:
I think the actual user data would be widely used in the future but okay to use current way for know.